include/internal/ktls.h

```diff
@@ -119,6 +205,108 @@ static ossl_inline ossl_ssize_t ktls_sendfile(int s, int fd, off_t off,
     }
     return sbytes;
 }
+
+#   ifdef OSSL_SSL_LOCAL_H
+/*-
+ * Check if a given cipher is supported by the KTLS interface.
+ * The kernel might still fail the setsockopt() if no suitable
+ * provider is found, but this checks if the socket option
+ * supports the cipher suite used at all.
+ */
+static ossl_inline int ktls_check_supported_cipher(SSL *s)
+{
+
+    switch (s->version) {
+    case TLS1_VERSION:
+    case TLS1_1_VERSION:
+    case TLS1_2_VERSION:
+    case TLS1_3_VERSION:
+        break;
+    default:
+        return 0;
+    }
+
+    switch (s->s3.tmp.new_cipher->algorithm_enc) {
+    case SSL_AES128GCM:
+    case SSL_AES256GCM:
+        return 1;
+    case SSL_AES128:
+    case SSL_AES256:
+        if (s->ext.use_etm)
```

Comment:
KTLS only does mac-then-encrypt????

Reply:
Yes.  One of my clients builds a NIC which supports TLS offload using a TCP offload engine, and for AES-CBC cipher suites it only supports MTE.  One of my other clients is Netflix and what they have found is that any of their clients that are new enough to use MTE are also new enough to just use AES-GCM.  Only older, legacy clients use AES-CBC cipher suites and those are all using MTE.  I could add ETM support to the KTLS layer in FreeBSD pretty easily, but there hasn't been any demand for it.

Reply:
Ok - fair enough.